### PR TITLE
Update matrix-bot-sdk to 0.6.0

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes https://github.com/matrix-org/matrix-hookshot/issues/435

This brings us up to date, and also unlocks the encryption support in a later PR. It also makes us require Node 16.

~~Blocked by requiring Node 16, and https://github.com/turt2live/matrix-bot-sdk/issues/247~~

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "jira-client": "^8.0.0",
     "markdown-it": "^12.3.2",
     "matrix-appservice-bridge": "^4.0.1",
-    "matrix-bot-sdk": "^0.5.19",
+    "matrix-bot-sdk": "^0.6.0",
```
Comment (AndrewFerr) [inline package.json:56]: Since https://github.com/turt2live/matrix-bot-sdk/issues/247 was fixed, this can use `^0.6.1` now, right?

Comment (Half-Shot) [reply package.json]: Done!

File: src/Connections/GitlabIssue.ts [hunk 2]
```diff
@@ -44,8 +44,8 @@ export class GitLabIssueConnection extends BaseConnection implements IConnection
     static readonly QueryRoomRegex = /#gitlab_(.+)_(.+)_(\d+):.*/;
     static readonly ServiceCategory = "gitlab";
 
-    static getTopicString(authorName: string, state: string) {
```
Comment (tadzik) [inline src/Connections/GitlabIssue.ts:47]: Huh, (how) did this work before? Do we even use this?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -48,14 +48,14 @@ jobs:
         uses: actions/setup-node@v1
         with:
           node-version: 16
-      - run: yarn --ignore-scripts
+      - run: yarn # Need to build to get rust bindings
```
Comment (tadzik) [inline .github/workflows/main.yml:51]:
```suggestion
      - run: yarn # Need to build scripts to get rust bindings
```

Just to make it more obvious :)

**Review by tadzik — APPROVED** [review]

File: src/Connections/GitlabIssue.ts (hunk 2 above)
Comment (Half-Shot) [reply src/Connections/GitlabIssue.ts:47]: Don't think this was ever used :disappointed:

Comment (tadzik) [reply src/Connections/GitlabIssue.ts:47]: I guess it's time to drop it then :)

**Pull request merged**